test/integration/auth.spec.ts

```diff
@@ -833,53 +847,120 @@ describe('admin.auth', () => {
       });
   });
 
-  it('verifyIdToken() fails when called with an invalid token', () => {
-    return admin.auth().verifyIdToken('invalid-token')
-      .should.eventually.be.rejected.and.have.property('code', 'auth/argument-error');
-  });
+  describe('verifyIdToken()', () => {
+    const uid = generateRandomString(20).toLowerCase();
+    const email = uid + '@example.com';
+    const password = 'password';
+    const userData = {
+      uid,
+      email,
+      emailVerified: false,
+      password,
+    };
+
+    // Create the test user before running this suite of tests.
+    before(() => {
+      return admin.auth().createUser(userData);
+    });
+
+    // Sign out after each test.
+    afterEach(() => {
+      return clientAuth().signOut();
+    });
+
+    after(() => {
+      return safeDelete(uid);
+    });
+
+    it('verifyIdToken() fails when called with an invalid token', () => {
+      return admin.auth().verifyIdToken('invalid-token')
+        .should.eventually.be.rejected.and.have.property('code', 'auth/argument-error');
+    });
 
-  if (authEmulatorHost) {
-    describe('Auth emulator support', () => {
-      const uid = 'authEmulatorUser';
-      before(() => {
-        return admin.auth().createUser({
-          uid,
-          email: 'lastRefreshTimeUser@example.com',
-          password: 'p4ssword',
+    it('verifyIdToken() fails with checkRevoked set to true and corresponding user disabled', () => {
+      let currentIdToken: string;
+      let currentUser: User;
+      return clientAuth().signInWithEmailAndPassword(email, password)
+        .then(({ user }) => {
+          expect(user).to.exist;
+          expect(user!.email).to.equal(email);
+          currentUser = user!;
+          return user!.getIdToken();
+        })
+        .then((idToken) => {
+          currentIdToken = idToken;
+          return admin.auth().verifyIdToken(currentIdToken, true);
+        })
+        .then((decodedIdToken) => {
+          expect(decodedIdToken.uid).to.equal(uid);
+          expect(decodedIdToken.email).to.equal(email);
+          return admin.auth().updateUser(uid, { disabled: true });
+        })
+        .then((userRecord) => {
+          // Ensure disabled field has been updated.
+          expect(userRecord.uid).to.equal(uid);
+          expect(userRecord.email).to.equal(email);
+          expect(userRecord.disabled).to.equal(true);
+          return admin.auth().verifyIdToken(currentIdToken, false);
+        })
+        .then(() => {
```

Comment:
This test passed for me with this block commented out (i.e. no reload). I added some extra logging to the implementation to log the user record every time the token is verified. Here's my output:

```
> mocha test/integration/*.ts --slow 5000 --timeout 20000 --require ts-node/register



  admin.auth
    verifyIdToken()
{"uid":"2qvksybtzo8ecdnvu0de","email":"2qvksybtzo8ecdnvu0de@example.com","emailVerified":false,"disabled":false,"metadata":{"lastSignInTime":"Tue, 10 Aug 2021 18:11:43 GMT","creationTime":"Tue, 10 Aug 2021 18:11:43 GMT"},"tokensValidAfterTime":"Tue, 10 Aug 2021 18:11:43 GMT","providerData":[{"uid":"2qvksybtzo8ecdnvu0de@example.com","email":"2qvksybtzo8ecdnvu0de@example.com","providerId":"password"}]}
{"uid":"2qvksybtzo8ecdnvu0de","email":"2qvksybtzo8ecdnvu0de@example.com","emailVerified":false,"disabled":true,"metadata":{"lastSignInTime":"Tue, 10 Aug 2021 18:11:43 GMT","creationTime":"Tue, 10 Aug 2021 18:11:43 GMT"},"tokensValidAfterTime":"Tue, 10 Aug 2021 18:11:43 GMT","providerData":[{"uid":"2qvksybtzo8ecdnvu0de@example.com","email":"2qvksybtzo8ecdnvu0de@example.com","providerId":"password"}]}
      ✓ verifyIdToken() fails with checkRevoked set to true and corresponding user disabled


  1 passing (7s)
```

May be we can try commenting this block out, and pushing the change to GitHub? I've enabled staging on this PR, so future pushes to this PR will trigger integration tests (note: you will also have to remove the `currentUser` local variable if you comment out this block).

Reply:
looks like it doesn't work as expected